broken installation with astropy v5

I am trying to install it, but there seems to be an important issue with the newer `astropy`. 

I cloned the `casa-formats-io` repo, and tried to install it with both `pip` and `python setup.py` and got a similar error message.  See below.

```
python setup.py develop did not run successfully.
â exit code: 1
â°â> [36 lines of output]
    running develop
    /private/var/folders/2n/6gk_yf0d1zvdwj6wy12h_klr0000gn/T/pip-build-env-pidylk7f/overlay/lib/python3.9/site-packages/setuptools/command/easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
      warnings.warn(
    /private/var/folders/2n/6gk_yf0d1zvdwj6wy12h_klr0000gn/T/pip-build-env-pidylk7f/overlay/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
      warnings.warn(
    running egg_info
    writing casa_formats_io.egg-info/PKG-INFO
    writing dependency_links to casa_formats_io.egg-info/dependency_links.txt
    writing entry points to casa_formats_io.egg-info/entry_points.txt
    writing requirements to casa_formats_io.egg-info/requires.txt
    writing top-level names to casa_formats_io.egg-info/top_level.txt
    reading manifest template 'MANIFEST.in'
    warning: no files found matching 'README.md'
    warning: no files found matching 'LICENSE.rst'
    warning: no files found matching '*.c' under directory '*.pyx'
    warning: no files found matching '*.pxd' under directory '*.pyx'
    warning: no files found matching '*' under directory 'licenses'
    warning: no files found matching '*' under directory 'cextern'
    warning: no files found matching '*' under directory 'scripts'
    no previously-included directories found matching 'build'
    no previously-included directories found matching 'docs/_build'
    no previously-included directories found matching 'docs/api'
    warning: no previously-included files matching '*.pyc' found anywhere in distribution
    warning: no previously-included files matching '*.o' found anywhere in distribution
    adding license file 'LICENSE'
    writing manifest file 'casa_formats_io.egg-info/SOURCES.txt'
    running build_ext
    building 'casa_formats_io._casa_chunking' extension
    x86_64-apple-darwin13.4.0-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/jpineda/miniconda3/envs/py3/include -fPIC -O2 -isystem /Users/jpineda/miniconda3/envs/py3/include -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/jpineda/miniconda3/envs/py3/include -D_FORTIFY_SOURCE=2 -isystem /Users/jpineda/miniconda3/envs/py3/include -I/private/var/folders/2n/6gk_yf0d1zvdwj6wy12h_klr0000gn/T/pip-build-env-pidylk7f/overlay/lib/python3.9/site-packages/numpy/core/include -I/Users/jpineda/miniconda3/envs/py3/include/python3.9 -c casa_formats_io/_casa_chunking.c -o build/temp.macosx-10.9-x86_64-cpython-39/casa_formats_io/_casa_chunking.o
    In file included from casa_formats_io/_casa_chunking.c:3:
    In file included from /Users/jpineda/miniconda3/envs/py3/include/python3.9/Python.h:91:
    /Users/jpineda/miniconda3/envs/py3/include/python3.9/bytearrayobject.h:9:10: fatal error: 'stdarg.h' file not found
    #include <stdarg.h>
             ^~~~~~~~~~
    1 error generated.
    error: command '/Users/jpineda/miniconda3/envs/py3/bin/x86_64-apple-darwin13.4.0-clang' failed with exit code 1
    [end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.
```

I am using python 3.9.10 and astropy 5.0.2. This is the version of GCC installed
```
(py3) bash-3.2$ gcc -v
Apple clang version 13.1.6 (clang-1316.0.21.2.3)
```
and the main python environment is managed with `conda`. What else would you like me to check? Because this is breaking both `SpectralCube` and `Turbustat` for me. :-( 

@keflavich @e-koch

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -10,11 +10,11 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.8, 3.9, 3.10]
         include:
           # Build in which CASA can be installed
           - os: ubuntu-18.04
-            python-version: 3.6
+            python-version: 3.9
     steps:
     - uses: actions/checkout@v2
     - name: Set up library dependencies

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -10,7 +10,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        python-version: [3.8, 3.9, 3.10]
+        python-version: [3.8, 3.9, '3.10']
         include:
           # Build in which CASA can be installed
           - os: ubuntu-18.04

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -20,7 +20,7 @@ jobs:
       - name: Build wheels
         run: python -m cibuildwheel --output-dir wheelhouse
         env:
-          CIBW_BUILD: cp36-* cp37-* cp38-* cp39-* cp310-*
+          CIBW_BUILD: cp38-* cp39-* cp310-*
           CIBW_TEST_EXTRAS: test
           CIBW_TEST_COMMAND: pytest --pyargs casa_formats_io
       - uses: actions/upload-artifact@v2

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -20,7 +20,7 @@ jobs:
       - name: Build wheels
         run: python -m cibuildwheel --output-dir wheelhouse
         env:
-          CIBW_BUILD: cp36-* cp37-* cp38-*
+          CIBW_BUILD: cp36-* cp37-* cp38-* cp39-* cp310-*
           CIBW_TEST_EXTRAS: test
           CIBW_TEST_COMMAND: pytest --pyargs casa_formats_io
       - uses: actions/upload-artifact@v2
@@ -35,7 +35,7 @@ jobs:
       - uses: actions/setup-python@v2
         name: Install Python
         with:
-          python-version: '3.7'
+          python-version: '3.9'
       - name: Install build
         run: python -m pip install build
       - name: Build sdist

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -14,7 +14,7 @@ jobs:
       - uses: actions/setup-python@v2
         name: Install Python
         with:
-          python-version: '3.7'
+          python-version: '3.9'
       - name: Install cibuildwheel
         run: python -m pip install cibuildwheel
       - name: Build wheels

--- a/.readthedocs.yml
+++ b/.readthedocs.yml
@@ -4,7 +4,7 @@ build:
   image: latest
 
 python:
-  version: 3.9
+  version: 3.8
   install:
     - method: pip
       path: .

--- a/.readthedocs.yml
+++ b/.readthedocs.yml
@@ -4,7 +4,7 @@ build:
   image: latest
 
 python:
-  version: 3.7
+  version: 3.9
   install:
     - method: pip
       path: .

--- a/README.rst
+++ b/README.rst
@@ -6,3 +6,6 @@ A small package implementing I/O for CASA data formats - see https://casa-format
 While developing this package, we made use of the https://github.com/casacore/casacore package to better understand
 the file format and have re-used some of the data class names. Therefore, since some parts of casa-formats-io may
 be considered translations of casacore into Python, we license casa-formats-io under the same LGPL license as casacore.
+
+
+casa-formats-io supports python versions >=3.8.

--- a/docs/index.rst
+++ b/docs/index.rst
@@ -17,6 +17,8 @@ to provide:
 At this time (November 2020), only reading .image datasets is supported. Reading measurement sets
 (.ms) or writing data of any kind are not yet supported.
 
+casa-formats-io supports python versions >=3.8.
+
 Using casa-formats-io
 ---------------------
 

--- a/setup.cfg
+++ b/setup.cfg
@@ -16,7 +16,7 @@ install_requires =
     astropy>=4.0
     numpy>=1.17
     dask[array]>=2.0
-python_requires = >=3.6
+python_requires = >=3.8
 
 [options.extras_require]
 test =

--- a/tox.ini
+++ b/tox.ini
@@ -1,6 +1,6 @@
 [tox]
 envlist =
-    py{36,37,38}-test{,-oldestdeps,-devdeps,-casa}
+    py{38,39,310}-test{,-oldestdeps,-devdeps,-casa}
     build_docs
     codestyle
 requires =

